src/utils.jl

```diff
@@ -1,14 +1,98 @@
 ## macro for argument checking
 
-macro check_args(D, cond)
-    quote
-        if !($(esc(cond)))
-            throw(ArgumentError(string(
-                $(string(D)), ": the condition ", $(string(cond)), " is not satisfied.")))
+macro strip_linenos(expr)
+    return esc(Base.remove_linenums!(expr))
+end
+
+"""
+    @check_args(
+        D,
+        @setup(statements...),
+        (arg₁, cond₁, message₁),
+        (cond₂, message₂),
+        ...,
+    )
+
+A convenience macro that generates AD-compatible checks of arguments for a distribution of
```

Comment:
I don't think one needs to say it generates AD-compatible checks. That's an implementation detail. You can just say it standardizes checking of arguments.

Reply:
Sure, I removed it now.